Another go at the coverage. Fixes #324

ninnghazad opened the pull request:
- Changed lcov version in .travis.yml
- Changed lcov usage in .travis.yml
- Changed CMakeLists to use --coverage.

A cool 99.266%:
https://coveralls.io/builds/24315289

ninnghazad commented:
#324
Let's see if this still works as a PR.

ninnghazad commented:
AppVeyor?? What is this now... Did i just fix the wrong CI?
I'm going to bed.

louis-langholtz commented on PlayRho/Common/Math.hpp:
```diff
@@ -392,9 +392,6 @@ PLAYRHO_CONSTEXPR inline auto Dot(const T1 a, const T2 b) noexcept
 ///
 /// @sa https://en.wikipedia.org/wiki/Cross_product
 ///
-/// @param a Value A of a 2-element type.
```
Hmm... did I make this change? LOLs! Wondering now why I did this. I'm imagining this relates to the doxygen build check that's run in CI. But I'd think removing these doxygen comments would cause this part of the build to have an issue instead of fixing an issue with it. I guess it can always be added back in later easily enough if needed.

ninnghazad commented:
I've added another branch to my fork based on master where i squashed the commits above: https://github.com/ninnghazad/PlayRho/commits/fix-travis-another
maybe that would be cleaner. the combined commit would be:
https://github.com/ninnghazad/PlayRho/commit/f4b60a5417ff5b319c08f7a8d0b20aee6dfabccc
let me know if that was what you are going for - or if you insist on using cpp-coveralls over coveralls-lcov. i took it that you were just looking for away to get coverage back on track, no matter the tool.

louis-langholtz commented:
@ninnghazad Thank you for all your work on this! I really appreciate it; it's been super helpful! I just did a squash merge of this pull request. Regarding which coverage tool(s) to use, if you have better experience with something other than what PlayRho is using, I'd love to consider using that alternative. I'd just picked the coverage tool set that it's using based on what github/coveralls.io seemed to be suggesting.
